eng/tools/generate-doc/index.ts

```diff
@@ -8,9 +8,27 @@ import { loadTheme } from "./theme/theme.js";
 async function runTypeDoc({ outputFolder, cwd }: { outputFolder: string; cwd: string }) {
   const oldCwd = process.cwd();
   process.chdir(cwd);
+  const packageJsonPath = path.join(cwd, "package.json");
+  const content = await readFile(packageJsonPath, "utf-8");
+  const packageJson = JSON.parse(content);
+  let entryPoints = ["src/index.ts"];
+  const exports = packageJson["exports"];
+  if (exports) {
+    entryPoints = [];
+    for (const [key, value] of Object.entries(exports)) {
+      if (key === "./package.json") {
+        continue;
+      }
+      const sourcePath = (value as Record<string, any>)["import"]?.["default"];
+      if (sourcePath) {
+        entryPoints.push(sourcePath.replace("./dist/esm", "src").replace(".js", ".ts"));
+      }
+    }
+  }
+  console.log(`  entrypoints: ${JSON.stringify(entryPoints)}`);
```

Comment:
nitpick: Is the leading space intentional?
```suggestion
  console.log(`entrypoints: ${JSON.stringify(entryPoints)}`);
```